packages/components/src/tooltip/test/index.js

```diff
@@ -261,111 +217,131 @@ describe( 'Tooltip', () => {
 		} );
 
 		it( 'should not show tooltip if the mouse leaves the anchor before the tooltip has shown', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
-
-			const MOUSE_LEAVE_DELAY = TOOLTIP_DELAY - 200;
+			const user = userEvent.setup();
 			const onMouseEnterMock = jest.fn();
 			const onMouseLeaveMock = jest.fn();
+			const MOUSE_LEAVE_DELAY = TOOLTIP_DELAY - 200;
 
 			render(
 				<>
-					<Tooltip text="Help text">
-						<button
+					<Tooltip { ...props }>
+						<Button
 							onMouseEnter={ onMouseEnterMock }
 							onMouseLeave={ onMouseLeaveMock }
 						>
 							Hover Me!
-						</button>
+						</Button>
 					</Tooltip>
-					<button>Hover me instead!</button>
+					<Button>Hover me instead!</Button>
 				</>
 			);
 
-			const externalButton = screen.getByRole( 'button', {
-				name: 'Hover me instead!',
-			} );
-			const tooltipButton = screen.getByRole( 'button', {
-				name: 'Hover Me!',
-			} );
-
-			await user.hover( tooltipButton );
+			await user.hover(
+				screen.getByRole( 'button', {
+					name: 'Hover Me!',
+				} )
+			);
 
 			// Tooltip hasn't appeared yet
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
 			expect( onMouseEnterMock ).toHaveBeenCalledTimes( 1 );
 
 			// Advance time by MOUSE_LEAVE_DELAY time
-			act( () => jest.advanceTimersByTime( MOUSE_LEAVE_DELAY ) );
+			await waitFor(
+				() =>
+					expect(
+						screen.queryByText( 'tooltip text' )
+					).not.toBeInTheDocument(),
+				{ timeout: MOUSE_LEAVE_DELAY }
+			);
```

Comment:
I wonder if this assertion is actually testing what we want.

Since the tooltip shouldn't be visible , this `waitFor` call would possibly immediately "pass", instead of performing the check at the end of the timeout time.

Basically, using `waitFor` works well when waiting for a condition to happen asynchronously — but it's likely not the right tool to test for a condition NOT to change for a specific amount of time.

@tyxla , do you have any thoughts? If my assumptions are correct, we may have to rewrite a few assertions throughout this file.

Reply:
Yep, I've taken a look, and it seems like `timeout` doesn't work the way the current version of the assertion assumes.

The timeout works like this: it will run on every X ms (X being the specified `interval`, 50ms by default), and if the assertion passes on the first run, it will resolve.

That means that we should instead actually set a timeout, and perform an assertion after that timeout.

Alternatively, we could use [`waitForElementToBeRemoved()`](https://testing-library.com/docs/dom-testing-library/api-async/#waitforelementtoberemoved), which should better express what we're aiming to do here, without having to specifically play with setting up intervals and timeouts.

Reply:
Thank you both! That makes sense why `waitFor` doesn't work well in this situation, where we are testing that something shouldn't happen after a set time. 

However, I am unsure if I understand why `waitForElementToBeRemoved()` would work in this case. The tooltip should never be rendered because the mouse moved away from the anchor before it's set `delay`. So how would waiting for it to be removed from the DOM work here?

Reply:
Sounds great, thank you!